src/main/java/net/neoforged/neoforge/common/conditions/FlagCondition.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.common.conditions;
+
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.MapCodec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+import net.minecraft.world.flag.FeatureFlag;
+import net.minecraft.world.flag.FeatureFlagSet;
+import net.minecraft.world.flag.FeatureFlags;
+
+/**
+ * Condition checking for the enabled state of a given {@link FeatureFlagSet}.
+ * <p>
+ * {@code requiredFeatures} - {@link FeatureFlagSet} containing all {@link FeatureFlag feature flags} to be validated.
+ * {@code checkEnabled} - Validates that all given {@link FeatureFlag feature flags} are enabled when {@code true} or disabled when {@code false}.
+ *
+ * @apiNote Mainly to be used when flagged content is not contained within the same feature pack which also enables said {@link FeatureFlag feature flags}.
+ */
+public final class FlagCondition implements ICondition {
+    public static final MapCodec<FlagCondition> CODEC = RecordCodecBuilder.mapCodec(instance -> instance.group(
+            FeatureFlags.CODEC.fieldOf("flags").forGetter(condition -> condition.requiredFeatures),
+            Codec.BOOL.lenientOptionalFieldOf("check_enabled", true).forGetter(condition -> condition.checkEnabled)).apply(instance, FlagCondition::new));
+
+    private final FeatureFlagSet requiredFeatures;
+    private final boolean checkEnabled;
+
+    private FlagCondition(FeatureFlagSet requiredFeatures, boolean checkEnabled) {
+        this.requiredFeatures = requiredFeatures;
+        this.checkEnabled = checkEnabled;
+    }
+
+    @Override
+    public boolean test(IContext context) {
+        var flagsEnabled = requiredFeatures.isSubsetOf(context.enabledFeatures());
+        // true if: 'checkEnabled' is true nd all given flags are enabled
+        // false if: `enabledEnabled' is false and all given flags are disabled
+        return flagsEnabled == checkEnabled;
```

Comment:
This does not line up with the comments. I interpret the checkEnabled field as a conditional so in logical expression `!checksEnabled OR flagsEnabled`, if that is not the case then the documentation should be updated on the class header, alternatively this commend needs to be addressed or the equivalency check modified.

Reply:
Based on code below, a modification to the field name might be appropriate? Given that checkEnabled is supposedly to indicate whether the check passed or not, not whether the check is even enabled at all.

Reply:
the `checkEnabled` bool is supposed to change whether the flags need to be enabled or not
- if true the condition should pass when all flags are enabled
- if false the condition should pass when all flags are disabled
I could be wrong on the bool logic here, it was slightly more complex but idea said it could be simplified to this and thats what i used

Reply:
Renamed field as that is a much better name i must admit (naming fields is hard) and implemented test to ensure the condition logic works as expected.